### DISCUSSION TITLE
Add field to Audit log tables to show source of action

### DISCUSSION THREAD

**Discussion by lexiclevenger** [specify/specify7#5956] [Feature Requests]
> Recent data investigations have made it clear that it would be very helpful when looking at the logs to know how the data was entered into the system. Would it be possible to track whether a create/edit/delete was done via a form or via a Workbench dataset?
> In addition it would be good to include the bulk edit tool when that is available, and possibly if the changes occurred as part of an external API call?

**Reported By**
Dan Baker at CSIRO on [Asana](https://app.asana.com/0/1201972579963460/1207452653273525/f)

**Comment by realVinayak** [comment]
hmm they can correlate dataset uploaded time with auditlogs? if timestampcreated are very close to each other (at level of seconds), it is almost always a wbdataset

**Comment by danb213** [comment]
It's very time-intensive for us to do that, and in the event of a user performing a rollback and re-upload there can be quite a large de-sync if there are bugs to search for. We are also very keen to track the usage of the bulk edit tool when that becomes available since that has the potential to create bigger problems across more data, so tracking that down to a single user/event becomes very useful for the Data Management team
